hibernate-reactive-core/src/test/java/org/hibernate/ReactiveStatelessDefaultBatchSizeTest.java

```diff
@@ -0,0 +1,466 @@
+ */
+package org.hibernate;
+
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.BaseReactiveTest;
+import org.hibernate.reactive.stage.Stage;
+import org.hibernate.reactive.testing.SqlStatementTracker;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * The test aims to check that methods accepting the batch size as parameter e.g. {@link Stage.StatelessSession#insert(int, Object...)}
+ * work when {@link AvailableSettings.STATEMENT_BATCH_SIZE} hasn't been set.
+ */
+@Timeout(value = 10, timeUnit = MINUTES)
+public class ReactiveStatelessDefaultBatchSizeTest extends BaseReactiveTest {
+	private static SqlStatementTracker sqlTracker;
+
+	private static final String PIG_ONE_NAME = "One";
+	private static final String PIG_TWO_NAME = "Two";
+	private static final String PIG_THREE_NAME = "Three";
+	private static final String PIG_FOUR_NAME = "Four";
+	private static final String PIG_FIVE_NAME = "Five";
+	private static final String PIG_SIX_NAME = "Six";
+
+	private static final GuineaPig PIG_ONE = new GuineaPig( 11, PIG_ONE_NAME );
+	private static final GuineaPig PIG_TWO = new GuineaPig( 22, PIG_TWO_NAME );
+	private static final GuineaPig PIG_THREE = new GuineaPig( 33, PIG_THREE_NAME );
+	private static final GuineaPig PIG_FOUR = new GuineaPig( 44, PIG_FOUR_NAME );
+	private static final GuineaPig PIG_FIVE = new GuineaPig( 55, PIG_FIVE_NAME );
+	private static final GuineaPig PIG_SIX = new GuineaPig( 66, PIG_SIX_NAME );
+
+	private static final GuineaPig[] PIGS = { PIG_ONE, PIG_TWO, PIG_THREE, PIG_FOUR, PIG_FIVE, PIG_SIX, };
+
+	@Override
+	protected Set<Class<?>> annotatedEntities() {
+		return Set.of( GuineaPig.class );
+	}
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+
+		// Construct a tracker that collects query statements via the SqlStatementLogger framework.
+		// Pass in configuration properties to hand off any actual logging properties
+		sqlTracker = new SqlStatementTracker(
+				ReactiveStatelessDefaultBatchSizeTest::filter,
+				configuration.getProperties()
+		);
+		return configuration;
+	}
+
+	@BeforeEach
+	public void clearTracker() {
+		sqlTracker.clear();
+	}
+
+	@Override
+	protected void addServices(StandardServiceRegistryBuilder builder) {
+		sqlTracker.registerService( builder );
+	}
+
+	private static boolean filter(String s) {
+		String[] accepted = { "select", "insert ", "update ", "delete " };
```

Comment:
Why are you also including the `select`?
Basically, you are not excluding any log with this filter.

Reply:
removed it